crates/store/re_dataframe2/src/query.rs

```diff
@@ -625,27 +692,31 @@ impl QueryHandle<'_> {
                 .iter()
                 .flatten()
                 .flat_map(|streaming_state| {
-                    streaming_state
-                        .chunk
-                        .timelines()
-                        .values()
-                        // NOTE: Already took care of that one above.
-                        .filter(|time_column| *time_column.timeline() != self.query.filtered_index)
-                        // NOTE: Cannot fail, just want to stay away from unwraps.
-                        .filter_map(|time_column| {
-                            let cursor = streaming_state.cursor as usize;
-                            time_column
-                                .times_raw()
-                                .get(cursor)
-                                .copied()
-                                .map(TimeInt::new_temporal)
-                                .map(|time| {
-                                    (
-                                        *time_column.timeline(),
-                                        (time, time_column.times_array().sliced(cursor, 1)),
-                                    )
-                                })
-                        })
+                    match streaming_state {
+                        StreamingJoinState::StreamingJoinState(s) => s.chunk.timelines(),
+                        StreamingJoinState::Retrofilled(unit) => unit.timelines(),
```

Comment:
:melting_face: this one hurts my head to think about and has some weird edge cases.

I don't think we should use Retrofilled timestamps to update the values non-primary index columns.

Consider the following table:
```
T1 | T2 | X | Y
---+----+---+---
1  | 3  | A | D
2  | 2  | B | -
3  | 1  | C | -
```

The LatestAt value of T2 when T1=3 is clearly 1, however, my read of this logic is that when executing a LatestAt() on a view that included Y, we would pull forward the 3 Value and let it take precedence because we aggregate by max.

The only situations where it might make sense to do so would be a table that looked like:
```
T1 | T2 | X | Y
---+----+---+---
1  | 3  | A | D
2  | -  | B | -
3  | -  | C | -
```

But even then, if the table looked like:
```
T1 | T2 | X | Y
---+----+---+---
1  | 3  | A | D
2  | 2  | B | -
3  | -  | C | -
```
I would think the LatestAt value of T2 at T1 should still actually be 2, not 3.

Reply:
I think this is also related to: https://github.com/rerun-io/rerun/issues/7471

Reply:
Max aggregation answers the question "how old is the data that I'm looking at, in aggregate, on timeline Tn?".
The only way to see `Y=D` on `T2` is to move the time cursor to `3` and beyond.

Whether that's the semantics we want, I do not know. I don't think there is generalized right answer anyhow, only the end user knows how they expect these conflicts to be resolved.

I don't mind changing this at any point if we can specify a set of rules to follow.